DEVELOPER_GUIDE.md

```diff
@@ -422,6 +423,79 @@ Output plugins are defined in [flb_output.h](https://github.com/fluent/fluent-bi
+| Flags         | This field allows the user to set option flags. For example, it specifies in certain cases if multiple entries are allowed. |
+| Set Property  | This field decides if the property needs to be written to context or just validated. If the property needs to be written to context, the value of this field needs to `FLB_TRUE` or else the value will be `FLB_FALSE`.|
+| Offset        | This field represents the member offset. It is 0 if the property is not written to context and if the property is being written to context it is ```offsetof(struct name_of_plugin_structure, name_of_property)```. The macro offsetof() returns the offset of the field *member* from the start of the structure type.|
+| Description   | This field is so that the user can give a short description of the property. It is `NULL` if no description is needed or given. |
+
+For example for [stdout](https://github.com/fluent/fluent-bit/blob/master/plugins/out_stdout/stdout.c#L158) plugin the config map is something like :
+
+```c
+/* Configuration properties map */
+static struct flb_config_map config_map[] = {
+    {
+     FLB_CONFIG_MAP_STR, "format", NULL,
+     0, FLB_FALSE, 0,
+     NULL
+    },
+    {
+     FLB_CONFIG_MAP_STR, "json_date_key", "date",
+     0, FLB_TRUE, offsetof(struct flb_stdout, json_date_key),
+     NULL
+    },
+    /* Other Properties here */
+    /* EOF */
+    {0}
+};
+
+/* Plugin registration */
+struct flb_output_plugin out_stdout_plugin = {
+    ...
+    .config_map   = config_map
+};
+
+```
+In the above code snippet, the property *format* is a string of type string like json, msgpack etc. It has default value NULL(in which case it uses msgpack), no flags, and it is being only validated by the config map and hence set_property field is `FLB_FALSE` with member offset 0. No description is written for *format* property at present.
+Similarly, for the property *json_date_key*, type is string, default value is date, and it is being written to context so the set_property field is `FLB_TRUE` with a member offset. Again, no description is written for it.
+
+
+Upon initilization the engine loads the config map like [this](https://github.com/fluent/fluent-bit/blob/master/plugins/out_stdout/stdout.c#L48) :
+
+```c
+    ret = flb_output_config_map_set(ins, (void *) ctx);
+```
+
+[flb_output_config_map_set](https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_output.h#L510) returns [flb_config_map_set](https://github.com/fluent/fluent-bit/blob/master/src/flb_config_map.c#L513) which is a function used by plugins that needs to populate their context structure with the configuration properties already mapped.
+
+Some points to keep in mind while migrating an existing plugin to a config map interface :
+- All memory allocations and releases of properties on exit are handled by the config map interface.
+- Remove redundant parameter assignments that were used before introducing the config map.
```

Comment:
As someone who is fairly new to config maps, it's not clear to me what this means

Reply:
So, what I meant was prior to the introoduction of config maps the assignment of default values, exit and memory allocations of plugins were manually taken care of. But now they are taken care by the config maps interface. Also, on reexamining, I have already explained this in previous point so deleting this line.